tests/components/enphase_envoy/test_binary_sensor.py

```diff
@@ -0,0 +1,43 @@
+"""Test Enphase Envoy binary sensors."""
+
+from unittest.mock import AsyncMock
+
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.enphase_envoy.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+from tests.components.enphase_envoy import (
+    setup_with_selected_platforms,
+    validate_entity_count,
+)
+
+
+async def test_binary_sensor(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+    mock_envoy: AsyncMock,
+    test_parameters,
+) -> None:
+    """Test enphase_envoy binary_sensor entities."""
+    await setup_with_selected_platforms(hass, config_entry, [Platform.BINARY_SENSOR])
+
+    # number entities states should be created from test data
+    assert await validate_entity_count(hass, test_parameters, Platform.BINARY_SENSOR)
```

Comment:
I think it would be simpeler and clearer if we just did the `assert len(hass.states) == 4` here instead of the function, fixture and json file

Reply:
dat was een beetje dom van mij. Went for the wrong line to load fixture.